Marlin/src/lcd/TFTGLCD/marlinui_TFTGLCD.cpp

```diff
@@ -1008,12 +1008,42 @@ void MarlinUI::draw_status_screen() {
   }
 
   // Draw a generic menu item with pre_char (if selected) and post_char
-  void MenuItemBase::_draw(const bool sel, const uint8_t row, FSTR_P const fstr, const char pre_char, const char post_char) {
+  void MenuItemBase::_draw(const bool sel, const uint8_t row, FSTR_P const fstr, const char pre_char, const char post_char, const uint8_t style, const char *vstr, const uint8_t minFstr) {
     if (!PanelDetected) return;
     lcd_moveto(0, row);
+
     lcd.write(sel ? pre_char : ' ');
-    uint8_t n = lcd_put_u8str(fstr, itemIndex, itemStringC, itemStringF, LCD_WIDTH - 2);
-    for (; n; --n) lcd.write(' ');
+    uint8_t n = LCD_WIDTH - 2;
+    const bool center = bool(style & SS_CENTER), full = bool(style & SS_FULL);
+    const int8_t plen = fstr ? utf8_strlen(fstr) : 0,
+                 vlen = vstr ? utf8_strlen(vstr) : 0;
+    int8_t pad = (center || full) ? n - plen - vlen : 0;
+
+    // SS_CENTER: Pad with half of the unused space first
+    if (center) for (int8_t lpad = pad / 2; lpad > 0; --lpad) { lcd.write(' '); n--; }
+
+    // Draw as much of the label as fits
+    if (plen) {
+      const int8_t expl = n;
+      n = lcd_put_u8str(fstr, itemIndex, itemStringC, itemStringF, n);
+      pad -= (expl - n - plen); // Reduce the padding
```

Comment:
Could you explain the maths in the comment?
`n = lcd_put_u8str` looks suspicious. Should it be something like `n -= lcd_put_u8str...`?